services/galley/src/Galley/External/LegalHoldService.hs

```diff
@@ -148,3 +167,15 @@ validateServiceKey pem = liftIO $ readPublicKey >>= \pk ->
 
     minRsaKeySize :: Int
     minRsaKeySize = 256 -- Bytes (= 2048 bits)
+
+-- Types
+
+-- | When receiving tokens from other services which are 'just passing through'
+-- it's error-prone useless extra work to parse and render them from JSON over and over again.
+-- We'll just wrap them with this to give some level of typesafety and a reasonable JSON
+-- instance
```

Comment:
I think this is good for now, but I disagree with the "useless extra work to parse" part.  If we know it should be a legal hold device token, and those have internal structure, we should check that structure here before we pass it on internally.  This throws errors early on when they are still helpful.

Perhaps you can leave some sentence to this end in the haddocks?